packages/core/services/management.js

```diff
@@ -0,0 +1,366 @@
+        .filter(([file, created]) => Date.now() - created > offsetMs)
+        .forEach(([file, created]) => fs.rmSync(path.join(this.settings.exportDir, file), { force: true }));
+    },
+    /**
+     * Get all filenames of a directory and its subdirectories.
+     * @param {string} originalDirPath The path to query
+     * @param {string} dirPath For recursion only, current query path
+     * @param {string[]} filesArr For recursion only, accumulator for files already found
+     * @returns {string[]} Filenames relative to the queried directory
+     */
+    async getFilesInDir(originalDirPath, dirPath = originalDirPath, filesArr = []) {
+      if (!fs.existsSync(dirPath)) return [];
+      const files = fs.readdirSync(dirPath);
+
+      for (const file of files) {
+        const filePath = path.join(dirPath, file);
+        const stat = fs.statSync(filePath);
+
+        if (stat.isDirectory()) {
+          await this.getFilesInDir(originalDirPath, filePath, filesArr);
+        } else {
+          filesArr.push(path.relative(originalDirPath, filePath));
+        }
+      }
+
+      return filesArr;
+    },
+    async deleteDataset(dataset) {
+      // Delete dataset.
+      await this.broker.call('triplestore.dataset.delete', { dataset, iKnowWhatImDoing: true });
+    },
+    /**
+     * Create n-quad string from array of rdf-js quads or triples.
+     * @param {object[]} quads Array of quads to serialize
+     * @returns {string} Serialized n-quads string
+     */
+    async serializeQuads(quads) {
+      const quadStrings = [];
+      const serializer = new NTriplesSerializer();
+      const nQuadStream = serializer.import(Readable.from(quads));
+      nQuadStream.on('data', quadString => {
+        quadStrings.push(quadString);
+      });
+      // Wait until reading stream finished
+      await new Promise(resolve => {
+        nQuadStream.on('end', () => {
+          resolve();
+        });
+      });
+      return quadStrings.join('');
+    },
+    /**
+     * Create n-quads string from dataset and settings dataset of webId.
+     * @param {string} dataset Name of db dataset
+     * @param {string} webId WebId as registered in the settings
+     * @returns {string} n-quads serialized dump of dataset and settings records.
+     */
+    async createRdfDump(dataset, webId) {
+      const dumpQuery = `SELECT * { { ?s ?p ?o } UNION { GRAPH ?g { ?s ?p ?o } } }`;
+      /** @type {object[]} */
+      const datasetDump = await this.broker.call('triplestore.query', {
+        query: dumpQuery,
+        webId: 'system',
+        dataset,
+        accept: MIME_TYPES.JSON
+      });
+      const settingsDump = await this.broker.call('triplestore.query', {
+        dataset: this.settings.settingsDataset,
+        query: `SELECT * WHERE {
+          ?s ?p ?o .
+          FILTER EXISTS { ?s <http://semapps.org/ns/core#webId> "${webId}" }
+        }`,
+        accept: MIME_TYPES.JSON
+      });
```

Comment:
I made the settings export an optional parameter which defaults to `false`.
I can imagine it might be useful in some backup cases and this way it's more in the spirit of transparency towards users who want to receive _all_ information stored about them (more towards gdpr compliance).